rootfs/docker-entrypoint.sh

```diff
@@ -80,29 +80,47 @@ if [ "$1" = 'postgres' ]; then
 		EOSQL
 		echo
 
+		gosu postgres pg_ctl -D "$PGDATA" -m fast -w stop
+
 		echo
 		for f in /docker-entrypoint-initdb.d/*; do
 			case "$f" in
 				*.sh)  echo "$0: running $f"; . "$f" ;;
-				*.sql)
-					echo "$0: running $f";
-					psql -v ON_ERROR_STOP=1 --username "$POSTGRES_USER" --dbname "$POSTGRES_DB" < "$f"
-					echo
-					;;
 				*)     echo "$0: ignoring $f" ;;
 			esac
 			echo
 		done
 
-		gosu postgres pg_ctl -D "$PGDATA" -m fast -w stop
 		set_listen_addresses '*'
 
 		echo
 		echo 'PostgreSQL init process complete; ready for start up.'
 		echo
 	fi
 
-	exec gosu postgres "$@"
+	# spawn postgres in the background
+	gosu postgres "$@" &
+
+	SERVICE_PID=$!
+
+	# smart shutdown on SIGINT and SIGTERM
+	function on_exit() {
+		kill -TERM $SERVICE_PID
+		wait $SERVICE_PID 2>/dev/null
+		exit 0
+	}
+	trap on_exit INT TERM
+
+	# wait until postgres becomes available
+	until is_running >/dev/null 2>&1; do sleep 1; done
+
+	# Push a fresh backup so we have a new recovery baseline
+	gosu postgres envdir "$WALE_ENVDIR" wal-e backup-push "$PGDATA"
+
+	# Run periodic backups in the background
+	gosu postgres backup &
```

Comment:
I don't think /bin/backup can handle SIGTERM gently at this time (due to /bin/backup being an infinite `while` loop), though that is certainly something we should definitely consider in the future. Keep in mind that v1 did not cleanly shut down /bin/backup either and we didn't see any issues in production since the time to backup is very short and only occurs once every 12 hours by default, so the potential window of loss is very minimal. However, I'd be happy to add a ticket that which we can test, observe and gracefully handle SIGTERM in the middle of a base backup.

For reference, the SIGTERM handling code seen here [is code from Deis v1](https://github.com/deis/deis/blob/955dcea900d6822590c8bb01bb30d9fbf9190ac6/database/bin/boot#L103-L116) since postgres is now no longer PID 1 with this refactor. I'd love to get back to `exec`ing the main process so postgres can run as PID 1 for process management purposes, but given that we have multiple back-grounded processes running in the container that depend on each other (`postgres` and `/bin/backup`), I can't see a way around that. Without this crutch, the database would shut down immediately without forcing an immediate WAL log shipped to object storage, leaving us with a potential window of loss.